### PR TITLE
Show that fill_between and span_where provide similar functionalities.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

In preparation for a possible deprecation of BrokenBarHCollection (#24827), which may be a good idea.
Using axes methods is usually simpler for end users, anyways.

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: examples/lines_bars_and_markers/span_regions.py [hunk 1]
```diff
@@ -16,22 +14,21 @@
 
 t = np.arange(0.0, 2, 0.01)
 s1 = np.sin(2*np.pi*t)
-s2 = 1.2*np.sin(4*np.pi*t)
```
Comment (rcomer) [inline examples/lines_bars_and_markers/span_regions.py:19]: Since there is now no `s2`, `s1` could become simply `s`.

**Review by rcomer — APPROVED** [review]
Definite improvement.  I just have one trivial suggestion that you are welcome to ignore if you  wish.

**Comment by anntzer** [comment]
sure, done.

**Comment by QuLogic** [comment]
@meeseeksdev backport to v3.7.x

**Comment by QuLogic** [comment]
OK, will do it as part of the merge myself.

**Comment by QuLogic** [comment]
Actually, I misunderstood the conflict; this conflicts because it was deprecated in 3.7; there's no need to backport this PR.

**Pull request merged**